### PR TITLE
RUM-4372 Setup the CI and gradle tests for the new dd-sdk-android-trace-otel module

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

Adding the gitlab step for publishing the `dd-sdk-android-trace-otel` module
Removing the `dd-sdk-android-trace` unit tests exception, this was a leftover from a previous PR which I forgot to remove.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -72,8 +72,8 @@ tasks.register("checkAll") {
 }
 
 registerSubModuleAggregationTask("assembleLibraries", "assemble")
-registerSubModuleAggregationTask("unitTestRelease", "testReleaseUnitTest", exceptions = setOf("dd-sdk-android-trace"))
-registerSubModuleAggregationTask("unitTestDebug", "testDebugUnitTest", exceptions = setOf("dd-sdk-android-trace"))
```
Comment (0xnm) [inline build.gradle.kts:76]: there is no test task on the CI for Otel module?

Comment (mariusc83) [reply build.gradle.kts:76]: there is actually if you check that `registerSubModuleAggregationTask` it is doing this automatically for all the modules but let me double check in gitlab

Comment (0xnm) [reply build.gradle.kts:76]: then another question :) what changed so that it allows us to run tests without these exceptions?

Comment (mariusc83) [reply build.gradle.kts:76]: :), yes good question there, I think I added this exception when I ported the Java code as an intermediary change and I forgot to remove it :(. I checked the data I changed those lines and it make sense.

File: features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt [hunk 2]
```diff
@@ -135,50 +135,49 @@ class OtelTracerProvider(
...
-                                BuildConfig.OPENTELEMETRY_API_VERSION_NAME
-                        }
+            return executeIfJavaFunctionPackageExists(
+                sdkCore.internalLogger,
+                TracerProvider.noop()
+            ) {
+                val tracingFeature = sdkCore.getFeature(Feature.TRACING_FEATURE_NAME)
+                    ?.unwrap<Feature>()
+                val internalCoreWriterProvider = tracingFeature as? InternalCoreWriterProvider
+                if (tracingFeature == null || internalCoreWriterProvider == null) {
```
Comment (jonathanmos) [inline features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt:145]: checking whether tracingFeature is null may be redundant. tracingFeature can't be null without internalCoreWriterProvider also being

**Review by jonathanmos — DISMISSED** [review]

File: features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt (hunk 2 above)
Comment (mariusc83) [reply features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerProvider.kt]: not quite, you can have `tracingFeature` which is of type Feature but not implementing the `InternalCoreWrirerProvider`

**Review by xgouchet — APPROVED** [review]

**Pull request merged**